daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,621 @@
+package firewall
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/netip"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/daemon/device"
+	"github.com/NordSecurity/nordvpn-linux/meshnet"
+)
+
+var ErrRuleAlreadyActive = errors.New("this rule is already active")
+
+const (
+	iptables  = "iptables"
+	ip6tables = "ip6tables"
+)
+
+type IptablesExecutor interface {
+	ExecuteCommand(command string) error
+	ExecuteCommandIPv6(command string) error
+}
+
+type Iptables struct {
+	ip6tablesSupported bool
+}
+
+func AreIp6tablesSupported() bool {
+	// #nosec G204 -- input is properly sanitized
+	_, err := exec.Command(ip6tables, "-S").CombinedOutput()
+	return err != nil
+}
+
+func NewIptables() Iptables {
+	return Iptables{
+		ip6tablesSupported: AreIp6tablesSupported(),
+	}
+}
+
+func (i Iptables) ExecuteCommand(command string) error {
+	commandArgs := strings.Split(command, " ")
+
+	// #nosec G204 -- arg values are known before even running the program
+	if _, err := exec.Command(iptables, commandArgs...).CombinedOutput(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i Iptables) ExecuteCommandIPv6(command string) error {
+	if !i.ip6tablesSupported {
+		return errors.New("ip6tables are not supported")
+	}
+
+	commandArgs := strings.Split(command, " ")
+
+	// #nosec G204 -- arg values are known before even running the program
+	if _, err := exec.Command(ip6tables, commandArgs...).CombinedOutput(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type allowIncomingRule struct {
+	allowIncomingRule string
+	blockLANRules     []string
+}
+
+type PortRange struct {
+	min int
+	max int
+}
+
+type FirewallManager struct {
+	commandExecutor      IptablesExecutor
+	devices              device.ListFunc              // list network interfaces
+	allowIncomingRules   map[string]allowIncomingRule // peer public key to allow incoming rule
+	fileshareRules       map[string]string            // peers public key to allow fileshare rule
+	allowlistRules       []string
+	trafficBlockRules    []string
+	connmark             uint32
```

Comment:
No way to update this during a runtime. Either add a Set function or make this as an argument where needed

Reply:
As it is now, `connmark` is never updated during runtime. App tells the user to restart the daemon after configuring it. This implementation mostly tries to replicate functionality of the old implementation, just in a simpler way. Should it allow for dynamic `connmark` configuration?

Reply:
Well, we have a configuration for it so it should be handled during runtime without full daemon restart. Anything in between is on the table. Since this is likely never re-created, connmark will never be updated during runtime

Reply:
I guess there's also more than goes into connmark update than just firewall. By skimming the code, routing and domain rotator use it, neither can be dynamically updated. So the scope here is wider than just firewall.